pagopa/io-app
# chore: [IOCOM-1628] FIMS FastLogin handling

## Short description
This pr adds correct handling for possible FastLogin-related HTTP 401 responses in the FIMS auth flow.

https://github.com/user-attachments/assets/f80b19d6-4b63-40ce-9750-30301f76f2ee

## List of changes proposed in this pull request
- updated locales where necessary
- refactored abort/cancel FIMS utility saga to make it more modular and reusable (with some renaming when needed)
- added fastLogin failure check when the app could in fact receive an HTTP 401 response due to FastLogin (both fetch consents and open IAB sagas)
- small refactors here and there for readability
- addition of "idle" state in the SSO saga, in order to better differentiate the states and better handle loading states.
as of right now, the flow is such:

| STATE:             | idle                                           | consents                      | in-app-browser-loading       | abort                  | fastLogin_forced_restart                                                     |
|--------------------|------------------------------------------------|-------------------------------|------------------------------|------------------------|------------------------------------------------------------------------------|
| data               | none                                           | pot.Pot(consents)             | none                         | none                   | none                                                                         |
| isLoading          | no                                             | pot.isLoading                 | yes                          | yes                    | no                                                                           |
| how do we reach it | the flow has not been started, or has finished | messages CTA has been pressed | consents have been approved  | user has quit the flow | saga has been forced to restart by fastLogin, and will soon restart the flow |
| next action        | start flow                                     | approve consents              | show IAB and go back to idle | go back to idle        | restart flow>go back to consents                                             |


## How to test
using the dev-server, while checked out in the FIMS branch, modify the provider router to return 401 in `responseFromOAuthAuthorizeSecondInteraction` and in the handler for  `${baseProviderPath()}/interaction/:id` to return 401, and make sure the 401 failure is correctly handled.
Also, restart the dev-server mid-flow and make sure that the forced re-login does not break anything.
